From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Tue, 26 Jun 2012 19:28:14 +0200
Subject: Do not fail the /thread/thread4 test if prlimit() fails

This happens on the Debian buildds.

[smcv: Use g_test_skip_printf()]

Forwarded: no, Debian buildd specific
---
 glib/tests/thread.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/glib/tests/thread.c b/glib/tests/thread.c
index 9821894..17b838a 100644
--- a/glib/tests/thread.c
+++ b/glib/tests/thread.c
@@ -147,8 +147,11 @@ test_thread4 (void)
   nl.rlim_cur = 1;
 
   if (prlimit (getpid (), RLIMIT_NPROC, &nl, &ol) != 0)
-    g_error ("setting RLIMIT_NPROC to {cur=%ld,max=%ld} failed: %s",
-             (long) nl.rlim_cur, (long) nl.rlim_max, g_strerror (errno));
+    {
+      g_test_skip_printf ("setting RLIMIT_NPROC to {cur=%ld,max=%ld} failed: %s",
+                          (long) nl.rlim_cur, (long) nl.rlim_max, g_strerror (errno));
+      return;
+    }
 
   error = NULL;
   thread = g_thread_try_new ("a", thread1_func, NULL, &error);
